bitcoin/script/miniscript.h

```diff
@@ -750,7 +750,10 @@ struct Node {
 
 public:
     //! Return the size of the script for this expression (faster than ToString().size()).
-    size_t ScriptSize() const { return scriptlen; }
+    size_t GetScriptSize() const { return scriptlen; }
+
+    //! Check the size of the script against policy limits
+    bool CheckScriptSize() const { return GetScriptSize() <= MAX_STANDARD_P2WSH_SCRIPT_SIZE; }
```

Comment:
You never call it:
```diff
diff --git a/bitcoin/script/miniscript.h b/bitcoin/script/miniscript.h
index 04855fb..b03dc39 100644
--- a/bitcoin/script/miniscript.h
+++ b/bitcoin/script/miniscript.h
@@ -786,7 +786,7 @@ public:
     bool NeedsSignature() const { return GetType() << "s"_mst; }
 
     //! Do all sanity checks.
-    bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && CheckOpsLimit() && CheckStackSize() && CheckTimeLocksMix(); }
+    bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && CheckOpsLimit() && CheckStackSize() && CheckTimeLocksMix() && CheckScriptSize(); }
 
     //! Construct the script for this miniscript (including subexpressions).
     template<typename Ctx>
```

Reply:
Used the function in compiler checks. These functions would be used again when integrating with bitcoin core descriptors.

Reply:
Cool. I still think that it should be included to `IsSafeTopLevel` too though?

Reply:
It is right? In the same diff that you included?

Reply:
`git fetch sanket && git checkout sanket/timelocks && git grep CheckScriptSize` says otherwise :) My diff was to ask for you to include it

Reply:
Thanks, good catch :). I had many miniscript trees locally and I messed this up.